### ISSUE TITLE
Support filterable interface on RealmBaseAdapter

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-android-adapters#79]
Moved from https://github.com/realm/realm-java/issues/646

----
As pr. this discussion on Google Groups: 
https://groups.google.com/forum/#!topic/realm-java/KMwalH1sD3Y

This will require being able to move objects between threads. Will at least require https://github.com/realm/realm-java/issues/503 first
----

**Comment by Zhuinden** [comment]
This is a good example for using `Filter` and `Filterable` with RealmRecyclerViewAdapter

http://stackoverflow.com/questions/40630322/how-to-implement-filterable-in-realmrecyclerviewadapter/40632613#40632613

**Comment by Zhuinden** [comment]
No I think this should work as long as you are using `realm-android-adapters:3.0.0`

**Comment by maxbaldrighi** [comment]
I didn't notice the change in release. I was still using the 2.1.1.
Anyways, even after updating to 3.0.0 nothing...

Any idea/suggestion, guys?

**Comment by maxbaldrighi** [comment]
I really think that for me the problem is the `updateData()` method.

Looking at the source code I see this line RealmRecyclerViewAdapter.java, l. 205: `this.adapterData = data;`, within the `updateData()` method:

    public void updateData(@Nullable OrderedRealmCollection<T> data) {
        if (hasAutoUpdates) {
            if (isDataValid()) {
                //noinspection ConstantConditions
                removeListener(adapterData);
            }
            if (data != null) {
                addListener(data);
            }
        }

        this.adapterData = data;
        notifyDataSetChanged();
    }

When I first used RecyclerView I often forgot that changing the instance of the collection underneath the adapter was wrong, and I always used `addAll()` method to maintain the same collection instance and to keep receiving the UI updates from the `notify...()` methods of the adapter. 
I don't understand why this should behave differently from my rookie mistakes, since this line clearly replaces the original collection instance with a new one.

But I'm sure I'm missing something...

**Comment by Zhuinden** [comment]
You're using `getData()` inside `onBindViewHolder` right?

**Comment by maxbaldrighi** [comment]
Lifesaver!!!
Nope, I was using the collection passed as argument to the adapter, just like any other RecView adapter.

So, basically, what I didn't get is that with `RealmRecyclerViewAdapter`s I must always access the collection with `getData()`, right?

**Comment by Zhuinden** [comment]
Yeah, pretty much, otherwise you'll run into the problem that you won't access the new dataset after using `updateData()` (this might sound like a familiar problem 😛 )

**Comment by maxbaldrighi** [comment]
Again, thank you so much!
